fix(docs): use `onReceive` consistently as callback handler's 2nd argument

d4rekanguok opened the pull request:
Just a tiny error: Actor doc refers to callback actor's  2nd argument as `onReceive`, but it is `onEvent` in the linked entry about invoking callback ([link](https://xstate.js.org/docs/guides/communication.html#invoking-callbacks))

davidkpiano commented:
Thanks for this, it might be better to rename `onEvent` to `onReceive` since that better reflects what's happening.

d4rekanguok commented:
> it might be better to rename onEvent to onReceive since that better reflects what's happening.

I'm down to help out on this one if it's a better idea, it looks like it'd only involve changing the docs + typing name here?

https://github.com/davidkpiano/xstate/blob/771becf761c14319795685feda8b2f015a75707d/packages/core/src/types.ts#L195-L198

davidkpiano commented:
Yes, you can change the typing name.

d4rekanguok commented:
Update: I've changed all references (in docs, typing names and test) to the name of callback handler's arguments to `onReceive`.

The first argument, `callback`, is also referred to under different names (`sender`, `next`) in a few places so I changed those as well.
